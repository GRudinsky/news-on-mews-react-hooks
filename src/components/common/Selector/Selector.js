import React from 'react'

export default function Selector(props) {
 
  return (
    <>
      <p data-testid="title">{`${props.name}:`}</p>
      <select data-testid="selector" className="selector" onChange={props.onChange} >
        <option data-testid = "default-value">{props.defaultValue}</option>
        {props.data.map((item, idx) =>
          <option key={idx}>{item}</option>
        )}
      </select>
    </>
  )
}
