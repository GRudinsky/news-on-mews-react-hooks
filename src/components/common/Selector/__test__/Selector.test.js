import React from 'react'
import ReactDOM from 'react-dom'
import Selector from './../Selector'
import { render, cleanup } from '@testing-library/react'
import renderer from 'react-test-renderer'
import '@testing-library/jest-dom'

const countries=['us', 'gb', 'de']
const name = 'countries'

afterEach(cleanup)

it('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(<Selector data={countries}/>, div)
})

it('renders selector with title and options', () => {
  const { getByTestId } = render(<Selector data={countries} name={name}/>)
  expect(getByTestId('selector')).toHaveTextContent('gb')
  expect(getByTestId('title')).toHaveTextContent('countries')
})

it('calling render with the same component on the same container does not remount', () => {
  const { getByTestId, rerender } = render(<Selector data={countries} defaultValue={countries[0]} name={name} />)
  expect(getByTestId('default-value')).toHaveTextContent('us')
  expect(getByTestId('title')).toHaveTextContent('countries')

  // re-render the same component with different props
  rerender(<Selector data={countries} defaultValue={countries[1]} name={name}/>)
  expect(getByTestId('default-value')).toHaveTextContent('gb')
  expect(getByTestId('title')).toHaveTextContent('countries')
})

it("matches snapshot", () => {
  const component = renderer.create(<Selector name={name} data={countries} />)
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})