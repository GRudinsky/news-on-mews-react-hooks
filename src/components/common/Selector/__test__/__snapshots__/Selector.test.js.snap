// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Array [
  <p
    data-testid="title"
  >
    countries:
  </p>,
  <select
    className="selector"
    data-testid="selector"
  >
    <option
      data-testid="default-value"
    />
    <option>
      us
    </option>
    <option>
      gb
    </option>
    <option>
      de
    </option>
  </select>,
]
`;
